tools/build_file_generator/third_party_deps/third_party_deps_updater.sh

```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env bash
+
+# A script for updating third party dependencies inside third_party_deps_tool java_binary
+
+readonly REPO_ABS_PATH=$(git rev-parse --show-toplevel)
+# TODO: Make the getting of the path independent of the project
+readonly BUILD_FILE_ABS_PATH="$REPO_ABS_PATH/tools/build_file_generator/third_party_deps/BUILD"
+
+# Getting public third-party target names
+third_party_folder_path="$REPO_ABS_PATH/third_party"
+declare -a actual_third_party_targets
+for entry in $(bazel query 'attr("visibility", "//visibility:public", //third_party/...)'); do
+  actual_third_party_targets=("${actual_third_party_targets[@]}" "$entry")
+done
+
+# Deleting old targets
+sed -i '/# %START%/,/# %END%/{//!d}' ${BUILD_FILE_ABS_PATH}
```

Comment:
@val-fed
There are couple of concerns here:
- `sed`, to be honest, is pretty cryptic to read. Maybe explain in detail what `sed` is meant to do here in terms of text manipulation ("delete text between %START and %END, while preserving the markers"). This way we'll be able to think of a better way to achieve the same goal.
- `-i <command> <file>` will only work on GNU sed. To make it portable you should do `sed -i.bak -e command file && rm file.bak`. See [here](https://stackoverflow.com/questions/5694228/sed-in-place-flag-that-works-both-on-mac-bsd-and-linux).
- Also, your version doesn't work on BSD sed 😉. Please add a semicolon at the end so it does not fail with `extra characters at the end of d command`: 
```suggestion
sed -i '/# %START%/,/# %END%/{//!d;}' ${BUILD_FILE_ABS_PATH}
```